### PR TITLE
ibus-theme-tools: init at 4.2.0

### PR DESCRIPTION
**Pull request by HollowMan6**
###### Motivation for this change

Add IBus Theme Tools for Generating IBus Themes that can be used in #146353

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/ibus-theme-tools/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  disabled = python3Packages.pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "openSUSE";
+    repo = "IBus-Theme-Tools";
+    rev = "v${version}";
+    sha256 = "0i8vwnikwd1bfpv4xlgzc51gn6s18q58nqhvcdiyjzcmy3z344c2";
+  };
+
+  propagatedBuildInputs = [ python3Packages.tinycss2 python3Packages.pygobject3 gettext ];
```
Comment (jtojnar) [inline pkgs/tools/misc/ibus-theme-tools/default.nix:15]: Why propagate gettext?

File: pkgs/tools/misc/ibus-theme-tools/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{ lib, python3Packages, fetchFromGitHub, gettext }:
+
+python3Packages.buildPythonPackage rec {
```
Comment (jtojnar) [inline pkgs/tools/misc/ibus-theme-tools/default.nix:3]: Should likely be `buildPythonApplication`.

**Comment by HollowMan6** [comment]
@jtojnar Sorry for those simple mistakes...

File: pkgs/tools/misc/ibus-theme-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    rev = "v${version}";
+    sha256 = "0i8vwnikwd1bfpv4xlgzc51gn6s18q58nqhvcdiyjzcmy3z344c2";
+  };
+
+  BuildInputs = [ gettext ];
+
+  propagatedBuildInputs = with python3Packages; [ tinycss2 pygobject3 ];
+
+  # No test.
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ibus-theme-tools/default.nix:20]: Can we do a pythonImportsCheck instead?

```suggestion
  doCheck = false;

  pythonImportsCheck = [ "ibus_theme_tools" ];
```

File: pkgs/tools/misc/ibus-theme-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  disabled = python3Packages.pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "openSUSE";
+    repo = "IBus-Theme-Tools";
+    rev = "v${version}";
+    sha256 = "0i8vwnikwd1bfpv4xlgzc51gn6s18q58nqhvcdiyjzcmy3z344c2";
+  };
+
+  BuildInputs = [ gettext ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ibus-theme-tools/default.nix:15]:
```suggestion
  buildInputs = [ gettext ];
```

File: pkgs/tools/misc/ibus-theme-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
+{ lib, python3Packages, fetchFromGitHub, gettext }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ibus-theme-tools";
+  version = "4.2.0";
+  disabled = python3Packages.pythonOlder "3.6";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/ibus-theme-tools/default.nix:6]:
```suggestion
```
python3 always points to 3.9 or later as of writing.

**Comment by HollowMan6** [comment]
@SuperSandro2000 Thanks.

**Pull request merged**